clients/src/main/java/org/apache/kafka/clients/GroupRebalanceConfig.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.requests.JoinGroupRequest;
+
+import java.util.Optional;
+
+/**
+ * Class to extract group rebalance related configs.
+ */
+public class GroupRebalanceConfig {
+
+    public enum ProtocolType {
+        Consumer("consumer"),
+        Connect("connect");
+
+        private String type;
+
+        ProtocolType(String type) {
+            this.type = type;
+        }
+
+        public String get() {
+            return type;
```

Comment:
Also if you intend to have a 1-1 mapping between the enum name and the `String` that is returned here, you may as well skip the field and use:
`return name().toLowerCase(Locale.ROOT);`

Reply:
Some enum types do that (e.g. [ConnectorType](https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorType.java)). Some others use a `private final` field and the constructor ([TimestampType](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/record/TimestampType.java)). Up to you

Reply:
I would prefer first one for its cleanness